Kuadrant/limitador-operator
# Verbosity level

## What

Deploy limitador with specific verbosity level
Currently, only verbosity level 1 to 4 (both included) are allowed. Any verbosity level out of that range should be rejected

```yaml
apiVersion: limitador.kuadrant.io/v1alpha1
kind: Limitador
metadata:
  name: limitador-sample
spec:
  verbosity: 2
```
## Verification Steps

dev setup

```
make local-setup
```
Deploy the limitador CR with verbosity level set to `2`

```yaml
k apply -f - <<EOF
---
apiVersion: limitador.kuadrant.io/v1alpha1
kind: Limitador
metadata:
  name: limitador-sample
spec:
  verbosity: 2
EOF
```

Check deployment is correct

```
kubectl wait --timeout=300s --for=condition=Ready limitador limitador-sample
```
Should return

```
limitador.limitador.kuadrant.io/limitador-sample condition met
```

Check command line

```
kubectl get deployment/limitador-limitador-sample  -o yaml | yq '.spec.template.spec.containers[0].command'
```
should look like this

```yaml
- limitador-server
- -vv
- --http-port
- "8080"
- --rls-port
- "8081"
- /home/limitador/etc/limitador-config.yaml
- memory
```
> **Note**: `-vv` has been added to the command line
